Description: Don't use included clipper library
 Use system installed libpolyclipping instead of the one bundled with the
 source.
Forwarded: yes, by private email; likely will not be implemented upstream.
Last-Update: 2014-01-24

Index: cura-engine-14.01/Makefile
===================================================================
--- cura-engine-14.01.orig/Makefile	2014-01-23 13:57:29.000000000 -0500
+++ cura-engine-14.01/Makefile	2014-01-23 21:34:22.210147615 -0500
@@ -6,19 +6,20 @@
 #g++ main.cpp modelFile/modelFile.cpp clipper/clipper.cpp -I. -o CuraEngine
 
 CXX ?= g++
-CFLAGS += -I. -c -Wall -Wextra -O3 -fomit-frame-pointer
+CFLAGS += -I. -c -Wall -Wextra -O3 -fomit-frame-pointer `pkg-config --cflags polyclipping`
 # also include debug symbols
 #CFLAGS+=-ggdb
 LDFLAGS +=
+LDADD += `pkg-config --libs polyclipping`
 SOURCES  = bridge.cpp comb.cpp gcodeExport.cpp infill.cpp inset.cpp layerPart.cpp main.cpp optimizedModel.cpp pathOrderOptimizer.cpp polygonOptimizer.cpp raft.cpp settings.cpp skin.cpp skirt.cpp slicer.cpp support.cpp timeEstimate.cpp
-SOURCES += clipper/clipper.cpp modelFile/modelFile.cpp utils/gettime.cpp utils/logoutput.cpp
+SOURCES += modelFile/modelFile.cpp utils/gettime.cpp utils/logoutput.cpp
 OBJECTS = $(SOURCES:.cpp=.o)
 EXECUTABLE = ./CuraEngine
 UNAME := $(shell uname)
 
 ifeq ($(UNAME), Linux)
 	OPEN_HTML=firefox
-	LDFLAGS += --static
+	LDFLAGS +=
 endif
 ifeq ($(UNAME), Darwin)
 	OPEN_HTML=open
@@ -36,7 +37,7 @@
 all: $(SOURCES) $(EXECUTABLE)
 
 $(EXECUTABLE): $(OBJECTS)
-	$(CXX) $(LDFLAGS) $(OBJECTS) -o $@
+	$(CXX) $(LDFLAGS) $(OBJECTS) $(LDADD) -o $@
 
 .cpp.o:
 	$(CXX) $(CFLAGS) $< -o $@
Index: cura-engine-14.01/utils/intpoint.h
===================================================================
--- cura-engine-14.01.orig/utils/intpoint.h	2014-01-06 02:04:56.000000000 -0500
+++ cura-engine-14.01/utils/intpoint.h	2014-01-23 16:43:30.181062063 -0500
@@ -9,7 +9,7 @@
 #define INLINE static inline
 
 //Include Clipper to get the ClipperLib::IntPoint definition, which we reuse as Point definition.
-#include "clipper/clipper.hpp"
+#include <clipper.hpp>
 
 #include <limits.h>
 #include <stdint.h>
